src/query/ast/src/ast/statements/tabular_function.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+use std::fmt::Formatter;
+
+use crate::ast::write_comma_separated_list;
+use crate::ast::CreateTableSource;
+use crate::ast::Expr;
+use crate::ast::Identifier;
+use crate::ast::Query;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct CreateTabularFunctionStmt<'a> {
+    pub if_not_exists: bool,
+    pub name: Identifier<'a>,
+    pub args: Vec<Expr<'a>>,
```

Comment:
@BohuTANG Following snowflake doc the function arguments should be expression(may be column name or literal). I have some confusion here, I feel in definition these should be parameters which are just placeholder, why snowflake using arguments in definition? If here is arguments, what does the passing arguments do when calling this tabular function? Also if here is arguments, should I evaluate the expression in function definition plan or in function invocation plan?

Reply:
I noticed in createUDF struct, the parameters were parsed as Identifier, I also don't understand.

Reply:
> why snowflake using arguments in definition?
I guess it is for the data type check. Bigquery also has type:
https://cloud.google.com/bigquery/docs/reference/standard-sql/table-functions

Reply:
For your question, I would ping @sundy-li